### ISSUE TITLE
Self-references don't pull in environment variables

### ISSUE THREAD

**Issue by anujkumar93** [chimpler/pyhocon#213]
I am trying to accomplish a simple task which is to pull in an environment variable `MYVAR` into my generated pyhocon config. I have executed `export MYVAR=my_value` in my env.

I have the following in my `pyhocon_test.conf`:
```
MYVAR : ${MYVAR}
```
This raises the following error:
```
pyhocon.exceptions.ConfigSubstitutionException: Cannot resolve ${MYVAR}: (line: 1, col: 9). Check for cycles.
```

I know from the documentation that self-references work. Having the following in my conf file
```
MYVAR=2
MYVAR : ${MYVAR}
```
will give me the correct output `ConfigTree([('MYVAR', 2)])`.

I know environment variable substitutions also work. Having the following conf:
```
SECOND_VAR : ${MYVAR}
```
also gives the expected substituted output: `ConfigTree([('SECOND_VAR', 'my_value')])`.

I am unable to make them both work well together. My expected outcome is to pull in the environment variable without editing the key: `ConfigTree([('MYVAR', 'my_value')])`. I'm using `pyhocon==0.3.51` inside Python3.6 environment.

**Comment by ebuildy** [comment]
Not sure what you try to achieve, but to reference env variable, use ``${?MY_VAR}``

**Comment by anujkumar93** [comment]
@ebuildy I am trying to pull in an environment variable into my pyhocon config WITHOUT changing the key. If I have an env variable `MY_VAR`, I can't seem to port it as `MY_VAR` into my config.

```
MY_VAR=${?MY_VAR}``` will fail, and so will the other combinations I listed in my description above.

**Comment by ebuildy** [comment]
Ha I see better now (in your first post, there is no such syntax) and yes you are right, there is some issues with it.

I did a couple of tests, using https://github/ebuildy/tampis tool (``MY_VAR=toto ./tampis ./config.conf``), so:

```
MY_VAR=${?MY_VAR}
```

Gives ``{}`` (didnt work as you say)

```
MY_VAR2=${?MY_VAR}
```

Gives ``"MY_VAR2": "toto"`` (Works as excepted)

```
MY_VAR=${?MY_VAR}
MY_VAR2=${?MY_VAR}
```

Give parse error:

```
Traceback (most recent call last):
  File "./tampis", line 38, in <module>
    main()
  File "./tampis", line 22, in main
    confData = ConfigFactory.parse_file(args.config)
  File "/usr/local/lib/python3.7/site-packages/pyhocon/config_parser.py", line 99, in parse_file
    return cls.parse_string(content, os.path.dirname(filename), resolve, unresolved_value)
  File "/usr/local/lib/python3.7/site-packages/pyhocon/config_parser.py", line 150, in parse_string
    return ConfigParser().parse(content, basedir, resolve, unresolved_value)
  File "/usr/local/lib/python3.7/site-packages/pyhocon/config_parser.py", line 423, in parse
    has_unresolved = cls.resolve_substitutions(config, allow_unresolved)
  File "/usr/local/lib/python3.7/site-packages/pyhocon/config_parser.py", line 580, in resolve_substitutions
    cls._fixup_self_references(config, accept_unresolved)
  File "/usr/local/lib/python3.7/site-packages/pyhocon/config_parser.py", line 467, in _fixup_self_references
    for key in config:  # Traverse history of element
RuntimeError: OrderedDict mutated during iteration
```

**Comment by anujkumar93** [comment]
Fixed in #219
